### PR TITLE
Make it build with qt6

### PR DESCRIPTION
**Pull request by lewisspencer**
Make the lib build in qt6

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -34,12 +34,12 @@ endif()
...
 
 # Find the QtWidgets library
-find_package(Qt5 COMPONENTS
+find_package(Qt6 COMPONENTS
              Core
              Widgets
              Gui)
 
-qt5_add_resources(RESOURCES ./resources/resources.qrc)
+qt6_add_resources(RESOURCES ./resources/resources.qrc)
```
Comment (SteveHardingson) [inline CMakeLists.txt]:
```suggestion
qt_add_resources(RESOURCES ./resources/resources.qrc)
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -86,9 +86,9 @@ target_include_directories(nodes
 
 target_link_libraries(nodes
   PUBLIC
-    Qt5::Core
-    Qt5::Widgets
-    Qt5::Gui
+    Qt6::Core
+    Qt6::Widgets
+    Qt6::Gui
```
Comment (SteveHardingson) [inline CMakeLists.txt]:
```suggestion
    Qt::Core
    Qt::Widgets
    Qt::Gui
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -128,7 +128,7 @@ set_target_properties(nodes
 
 file(GLOB_RECURSE HEADERS_TO_MOC include/nodes/internal/*.hpp)
 
-qt5_wrap_cpp(nodes_moc
+qt6_wrap_cpp(nodes_moc
```
Comment (SteveHardingson) [inline CMakeLists.txt]:
```suggestion
qt_wrap_cpp(nodes_moc
```

**Review by SteveHardingson — CHANGES_REQUESTED** [review]
Minor changes here, to avoid us having to make some changes if Qt7 is ever a thing. I'd also recommend upping the minimum CMake version here, as it definitely isn't correct!

**Review by SteveHardingson — APPROVED** [review]

**Review by BenHetherington — APPROVED** [review]

**Pull request merged**